hx_lti_initializer/templatetags/hx_lti_initializer_extras.py

```diff
@@ -84,4 +84,21 @@ def assignment_object_exists(object_id, collection_id):
 		assignment = Assignment.objects.get(assignment_id=collection_id);
 		return True
 	except:
-		return False
\ No newline at end of file
+		return False
+	
+@register.simple_tag
+def get_annotation_by_id(id, students):
+	'''
+		Given the id of an annotation and a list of student_objects, this searches through
+		the student_objects and returns the text of the annotation with that id	
+	'''
+	# O(n^2) - maximum of students*student_annotations executions.
+	# TODO: Performance?
+	# alternative is a network request to get annotation by ID.
+	for student in students:
```

Comment:
Just curious, but from the *instructor_dashboard_view*, can we create a dictionary of annotations keyed by ID? We could then pass this dictionary to the tag, and instead of an O(n^2) search, it's just O(1) lookup. For small courses with not many students, this won't be a problem, but maybe we can avoid the issue altogether by creating a convenient data structure ahead of time in the view. Thoughts?